source/common/common/key_value_store_base.h

```diff
@@ -27,14 +30,18 @@ class KeyValueStoreBase : public KeyValueStore,
 
   std::string error;
   // KeyValueStore
-  void addOrUpdate(absl::string_view key, absl::string_view value) override;
+  void addOrUpdate(absl::string_view key, absl::string_view value,
+                   absl::optional<std::chrono::seconds> ttl) override;
+  // Removes all keys before calling flush()
```

Comment:
nit: this method should be moved out from under the `// KeyValueStore` comment, since it's specific to this class, not to the interface.